### DISCUSSION TITLE
macos 12.1    build failed

### DISCUSSION THREAD

**Discussion by taozhiwei21** [fltk-rs/fltk-rs#1135] [Q&A]
Error reporting information：

taozhiwei@taozhiweideMacBook-Pro fltk-rs % cargo build
taozhiwei@taozhiweideMacBook-Pro fltk-rs % cargo build
warning: Could not find invokable CMake or Git, building using the fltk-bundled feature flag!
warning: If this is not desirable, please ensure CMake and Git are installed and in your PATH!
   Compiling fltk-sys v1.2.30 (/Users/taozhiwei/fltk-rs/fltk-sys)
   Compiling fltk v1.2.30 (/Users/taozhiwei/fltk-rs/fltk)
error: could not find native static library `cfltk`, perhaps an -L flag is missing?

The following warnings were emitted during compilation:

warning: Could not find invokable CMake or Git, building using the fltk-bundled feature flag!
warning: If this is not desirable, please ensure CMake and Git are installed and in your PATH!

error: could not compile `fltk-sys` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed

Cmake and git were installed successfully。

**Comment by MoAlyousef** [comment]
Can both CMake and Git be invoked from the command-line?
What's the output of `$ cmake --version` and `$ git --version`.
I've updated the warning message in the master branch. If you had cloned fltk-rs, can you do a `git pull` then `cargo clean` and `cargo build -vv`?

The build script can’t find cmake or git, so it tries to download the prebuilt fltk lib, but there is no prebuilt bundle for arm macos.

**Reply by taozhiwei21** [reply]
It's done, thanks
